cms/navigation/wagtail_hooks.py

```diff
@@ -0,0 +1,33 @@
+from typing import TYPE_CHECKING
+
+from wagtail.permissions import ModelPermissionPolicy
+from wagtail.snippets.models import register_snippet
+from wagtail.snippets.views.snippets import SnippetViewSet
+
+from .models import MainMenu
+
+if TYPE_CHECKING:
+    from cms.users.models import User
+
+
+class NoAddModelPermissionPolicy(ModelPermissionPolicy):
+    """Model permission that doesn't allow creating more than one main menu instance."""
+
+    def user_has_permission(self, user: "User", action: str) -> bool:
+        if action == "add" and MainMenu.objects.exists():
+            return False
+        has_permission: bool = super().user_has_permission(user, action)
+        return has_permission
+
+
+class MainMenuViewSet(SnippetViewSet):
```

Comment:
Using the [Settings](https://docs.wagtail.org/en/v3.0.1/reference/contrib/settings.html) contrib app would negate the need for anything custom in this module... it only allows you to create one instance per site, and the UI is a little better, as it doesn't have the list view (which is kinda redundant)... Instead, if there are multiple sites, it just shows a 'site switcher' in the edit interface to switch to the entry for different sites.

Reply:
There are two considerations here:
1. we'd have potentially separate people managing the main and footer menus. (this could be satisfied with separate settings)
2. there is a need to preview the changes, and a push for a simple review workflow before the main menu (at a min) is updated.


Snippets cover both needs. And the navigation setting becomes a question of selecting the approviate instances.

Reply:
@zerolab Okay, good to know about the translatable aspect... that probably rules out settings. Is it worth thinking about multisite at this juncture, or do you think that's something to solve later if needed?

Reply:
I have been thinking about it, and `NavigationSettings` inherits from `BaseSiteSetting` so it should scale with it. At that point we can remove the artificial singleton `MainMenu` aspect